tsl/src/continuous_aggs/refresh.c

```diff
@@ -644,13 +648,25 @@ continuous_agg_refresh(PG_FUNCTION_ARGS)
 {
 	Oid cagg_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
 	bool force = PG_ARGISNULL(3) ? false : PG_GETARG_BOOL(3);
+	Jsonb *options = PG_ARGISNULL(4) ? NULL : PG_GETARG_JSONB_P(4);
+	bool process_hypertable_invalidations = true;
 	ContinuousAgg *cagg;
 	InternalTimeRange refresh_window = {
 		.type = InvalidOid,
 	};
 
 	ts_feature_flag_check(FEATURE_CAGG);
 
+	if (options)
+	{
+		bool found;
+		bool process_invalidations =
+			ts_jsonb_get_bool_field(options,
+									POL_REFRESH_CONF_KEY_PROCESS_HYPERTABLE_INVALIDATIONS,
+									&found);
+		process_hypertable_invalidations = !found || process_invalidations;
+	}
```

Comment:
An open question here is if we when validating the options should check for unrecognized options.

- If the user makes a typo, it might go unnoticed and lead to hard to find configuration errors. Checking for unrecognized options will then make this apparent.
- If we do not allow unrecognized options, there is no way for the user to add their own options used for other purposes, and it makes version handling more complicated.

A good middle ground could be to print a warning about unrecognized options.